sdk/core/System.ClientModel/src/Pipeline/MessageLoggingPolicy.cs

```diff
@@ -149,8 +149,16 @@ private async ValueTask ProcessSyncOrAsync(PipelineMessage message, IReadOnlyLis
                 }
                 else
                 {
-                    responseBytes = new byte[_maxLength];
-                    response.ContentStream.Read(responseBytes, 0, _maxLength);
```

Comment:
[CA2022: Avoid inexact read with Stream.Read](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2022)

This approach was more efficient because it avoids the extra allocation to a memory stream. However, it is unreliable because Read may return less bytes than maxLength, even if there are more than maxLength bytes in the stream. 

The updated approach using MaxLengthStream is what is done in Core's logging policy.